### ISSUE TITLE
Monit check produces warning in Zookeeper logs

### ISSUE THREAD

**Issue by tsindot** [rawmind0/alpine-zk#3]
First great set of docker images, great job.

The current monit check:
```
check process zk-service with pidfile /opt/zk/data/zookeeper_server.pid
  start program = "/opt/zk/bin/zk-service.sh start"
  stop program = "/opt/zk/bin/zk-service.sh stop"
  if failed port 2181 type tcp for 5 cycles then exec "/opt/monit/bin/monit quit"
```
Specifically the `if failed port 2181....`  produces the following WARN message in zookeeper logs:
```
2017-06-01 15:39:01,405 [myid:] - INFO  [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:2181:NIOServerCnxnFactory@192] - Accepted socket connection from /0:0:0:0:0:0:0:1:41038
2017-06-01 15:39:01,470 [myid:] - WARN  [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:2181:NIOServerCnxn@357] - caught end of stream exception
EndOfStreamException: Unable to read additional data from client sessionid 0x0, likely client has closed socket
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:228)
	at org.apache.zookeeper.server.NIOServerCnxnFactory.run(NIOServerCnxnFactory.java:203)
	at java.lang.Thread.run(Thread.java:745)
```

This does not cause an error just noise in the log.

A colleague of mine suggested converting this over to a custom protocol, which seems to have resolved the WARN message.  We modified it to do the following:
```
check process zk-service with pidfile /opt/zk/data/zookeeper_server.pid
  start program = "/opt/zk/bin/zk-service.sh start"
  stop program = "/opt/zk/bin/zk-service.sh stop"
  if failed port 2181 and send "ruok" expect "imok" for 5 cycles then exec "/opt/monit/bin/monit quit"
```
Log now looks like this:
```
2017-06-01 20:34:54,760 [myid:] - INFO  [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:2181:NIOServerCnxnFactory@192] - Accepted socket connection from /0:0:0:0:0:0:0:1:41596
2017-06-01 20:34:54,761 [myid:] - INFO  [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:2181:NIOServerCnxn@827] - Processing ruok command from /0:0:0:0:0:0:0:1:41596
2017-06-01 20:34:54,762 [myid:] - INFO  [Thread-30:NIOServerCnxn@1008] - Closed socket connection for client /0:0:0:0:0:0:0:1:41596 (no session established for client)
```

Will be happy to submit a PR if your interested.

**Comment by rawmind0** [comment]
Hi @tsindot , 

Sure i'm interested in this pretty cool PR!! Great job!!

Happy to merge...Thanks so much :)

**Comment by sinerxx** [comment]
Hi @tsindot , 
thx for help!
Log now looks like this: 

Sep 27 11:07:57 izbp1i3t9in0xp6xbaqmkgz bash[9235]: 2017-09-27 11:07:57,022 [myid:1] - INFO  [NIOServerCxnFactory.AcceptThread:/0.0.0.0:2181:NIOServerCnxnFactory$AcceptThread@296] - Accepted socket connection from /127.0.0.1:11146
Sep 27 11:07:57 izbp1i3t9in0xp6xbaqmkgz bash[9235]: 2017-09-27 11:07:57,022 [myid:1] - INFO  [NIOWorkerThread-8:NIOServerCnxn@627] - Closed socket connection for client /127.0.0.1:11146 (no session established for client)
Sep 27 11:07:59 izbp1i3t9in0xp6xbaqmkgz bash[9235]: 2017-09-27 11:07:59,522 [myid:1] - INFO  [NIOServerCxnFactory.AcceptThread:/0.0.0.0:2181:NIOServerCnxnFactory$AcceptThread@296] - Accepted socket connection from /127.0.0.1:11148
Sep 27 11:07:59 izbp1i3t9in0xp6xbaqmkgz bash[9235]: 2017-09-27 11:07:59,523 [myid:1] - INFO  [NIOWorkerThread-1:NIOServerCnxn@627] - Closed socket connection for client /127.0.0.1:11148 (no session established for client)
